### PR TITLE
Enable arm64 hyperv and SoCs support for CBL-Mariner

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Enable kernel configs to support Hyperv, SoCs like Cavium and Ampere for Arm64 CBL-Mariner.
Changes verified and confirm Mariner Arm64 images booted on HyperV, Cavium and Ampere SoCs.

###### Change Log  

- Enable kernel configs to support Hyperv, SoCs like Cavium and Ampere for Arm64 CBL-Mariner.

###### Does this affect the toolchain?  

YES
**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

**Review by jslobodzian — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/kernel/config_aarch64 [hunk 1]
```diff
@@ -956,32 +1070,36 @@ CONFIG_IP_PIMSM_V2=y
...
+CONFIG_NET_FOU_IP_TUNNELS=y
 CONFIG_INET_AH=m
 CONFIG_INET_ESP=m
-# CONFIG_INET_ESP_OFFLOAD is not set
+CONFIG_INET_ESP_OFFLOAD=m
 CONFIG_INET_IPCOMP=m
 CONFIG_INET_XFRM_TUNNEL=m
 CONFIG_INET_TUNNEL=m
-# CONFIG_INET_DIAG is not set
+CONFIG_INET_DIAG=m
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:1081]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 2]
```diff
@@ -2787,14 +3902,25 @@ CONFIG_VT_CONSOLE_SLEEP=y
...
 CONFIG_UNIX98_PTYS=y
-# CONFIG_LEGACY_PTYS is not set
-# CONFIG_SERIAL_NONSTANDARD is not set
-# CONFIG_NOZOMI is not set
-# CONFIG_N_GSM is not set
-# CONFIG_TRACE_SINK is not set
-# CONFIG_NULL_TTY is not set
-# CONFIG_LDISC_AUTOLOAD is not set
-# CONFIG_DEVMEM is not set
+CONFIG_LEGACY_PTYS=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:3905]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 3]
```diff
@@ -2787,14 +3902,25 @@ CONFIG_VT_CONSOLE_SLEEP=y
...
+CONFIG_SYNCLINK_GT=m
+CONFIG_NOZOMI=m
+CONFIG_ISI=m
+CONFIG_N_HDLC=m
+CONFIG_N_GSM=m
+CONFIG_TRACE_ROUTER=m
+CONFIG_TRACE_SINK=m
+CONFIG_NULL_TTY=m
+CONFIG_LDISC_AUTOLOAD=y
+CONFIG_DEVMEM=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:3923]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 4]
```diff
@@ -6916,6 +8559,8 @@ CONFIG_ARCH_HAS_UBSAN_SANITIZE_ALL=y
 # CONFIG_UBSAN is not set
 CONFIG_UBSAN_ALIGNMENT=y
 CONFIG_ARCH_HAS_DEVMEM_IS_ALLOWED=y
+CONFIG_STRICT_DEVMEM=y
+# CONFIG_IO_STRICT_DEVMEM is not set
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:8563]: This config has been identified by the kernel config hardening checker. This config will automatically be removed when CONFIG_DEVMEM is disabled

File: SPECS/kernel/config_aarch64 [hunk 5]
```diff
@@ -153,8 +160,8 @@ CONFIG_IPC_NS=y
 CONFIG_USER_NS=y
 CONFIG_PID_NS=y
 CONFIG_NET_NS=y
-# CONFIG_CHECKPOINT_RESTORE is not set
-# CONFIG_SCHED_AUTOGROUP is not set
+CONFIG_CHECKPOINT_RESTORE=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:163]: This config has been identified as attack surface from KSPP and was intentionally disabled earlier.

File: SPECS/kernel/config_aarch64 [hunk 6]
```diff
@@ -881,7 +987,8 @@ CONFIG_MMU_NOTIFIER=y
 CONFIG_KSM=y
 CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
 CONFIG_ARCH_SUPPORTS_MEMORY_FAILURE=y
-# CONFIG_MEMORY_FAILURE is not set
+CONFIG_MEMORY_FAILURE=y
+CONFIG_HWPOISON_INJECT=m
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:991]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 7]
```diff
@@ -2906,11 +4050,14 @@ CONFIG_TCG_ATMEL=m
...
-# CONFIG_TCG_VTPM_PROXY is not set
-# CONFIG_TCG_TIS_ST33ZP24_I2C is not set
-# CONFIG_TCG_TIS_ST33ZP24_SPI is not set
-# CONFIG_DEVPORT is not set
-# CONFIG_XILLYBUS is not set
+CONFIG_TCG_VTPM_PROXY=m
+CONFIG_TCG_TIS_ST33ZP24=m
+CONFIG_TCG_TIS_ST33ZP24_I2C=m
+CONFIG_TCG_TIS_ST33ZP24_SPI=m
+CONFIG_DEVPORT=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:4057]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 8]
```diff
@@ -589,31 +635,74 @@ CONFIG_ACPI_IPMI=m
 CONFIG_ACPI_HOTPLUG_CPU=y
 CONFIG_ACPI_THERMAL=y
 CONFIG_ARCH_HAS_ACPI_TABLE_UPGRADE=y
-# CONFIG_ACPI_TABLE_UPGRADE is not set
+CONFIG_ACPI_TABLE_UPGRADE=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:638]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 9]
```diff
@@ -540,18 +579,25 @@ CONFIG_ARM_PSCI_FW=y
...
+CONFIG_EFI_BOOTLOADER_CONTROL=m
+CONFIG_EFI_CAPSULE_LOADER=m
+CONFIG_EFI_TEST=m
 CONFIG_RESET_ATTACK_MITIGATION=y
 # end of EFI (Extensible Firmware Interface) Support
 
+CONFIG_UEFI_CPER=y
+CONFIG_UEFI_CPER_ARM=y
 CONFIG_EFI_EARLYCON=y
+CONFIG_EFI_CUSTOM_SSDT_OVERLAYS=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:600]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 10]
```diff
@@ -2787,14 +3902,25 @@ CONFIG_VT_CONSOLE_SLEEP=y
...
+CONFIG_SYNCLINKMP=m
+CONFIG_SYNCLINK_GT=m
+CONFIG_NOZOMI=m
+CONFIG_ISI=m
+CONFIG_N_HDLC=m
+CONFIG_N_GSM=m
+CONFIG_TRACE_ROUTER=m
+CONFIG_TRACE_SINK=m
+CONFIG_NULL_TTY=m
+CONFIG_LDISC_AUTOLOAD=y
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:3922]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 11]
```diff
@@ -540,18 +579,25 @@ CONFIG_ARM_PSCI_FW=y
...
+CONFIG_EFI_SOFT_RESERVE=y
 CONFIG_EFI_PARAMS_FROM_FDT=y
 CONFIG_EFI_RUNTIME_WRAPPERS=y
 CONFIG_EFI_ARMSTUB=y
 CONFIG_EFI_ARMSTUB_DTB_LOADER=y
-# CONFIG_EFI_CAPSULE_LOADER is not set
-# CONFIG_EFI_TEST is not set
+CONFIG_EFI_BOOTLOADER_CONTROL=m
+CONFIG_EFI_CAPSULE_LOADER=m
+CONFIG_EFI_TEST=m
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:593]: This config has been identified as attack surface from KSPP and was intentionally disabled earlier.

File: SPECS/kernel/config_aarch64 [hunk 12]
```diff
@@ -1346,39 +1496,107 @@ CONFIG_BRIDGE_EBT_REDIRECT=m
 CONFIG_BRIDGE_EBT_SNAT=m
 CONFIG_BRIDGE_EBT_LOG=m
 CONFIG_BRIDGE_EBT_NFLOG=m
-# CONFIG_BPFILTER is not set
-# CONFIG_IP_DCCP is not set
+CONFIG_BPFILTER=y
+CONFIG_BPFILTER_UMH=m
+CONFIG_IP_DCCP=m
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:1501]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 13]
```diff
@@ -2604,113 +3458,350 @@ CONFIG_B43_PHY_HT=y
...
+CONFIG_INPUT_MOUSEDEV=y
+CONFIG_INPUT_MOUSEDEV_PSAUX=y
 CONFIG_INPUT_MOUSEDEV_SCREEN_X=1024
 CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
-# CONFIG_INPUT_JOYDEV is not set
-CONFIG_INPUT_EVDEV=m
-# CONFIG_INPUT_EVBUG is not set
+CONFIG_INPUT_JOYDEV=m
+CONFIG_INPUT_EVDEV=y
+CONFIG_INPUT_EVBUG=m
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:3752]: This config has been identified by the kernel config hardening checker and should be disabled

File: SPECS/kernel/config_aarch64 [hunk 14]
```diff
@@ -671,11 +761,16 @@ CONFIG_HAVE_MOD_ARCH_SPECIFIC=y
 CONFIG_MODULES_USE_ELF_RELA=y
 CONFIG_ARCH_HAS_ELF_RANDOMIZE=y
 CONFIG_HAVE_ARCH_MMAP_RND_BITS=y
-CONFIG_ARCH_MMAP_RND_BITS=32
+CONFIG_ARCH_MMAP_RND_BITS=18
```
Comment (christopherco) [inline SPECS/kernel/config_aarch64:764]: This config has been identified by the kernel config hardening checker and should be set to 32.

File: SPECS/kernel/kernel.signatures.json [hunk 15]
```diff
@@ -1,7 +1,7 @@
 {
  "Signatures": {
   "config": "681e8ff274fc3fc79eaf38d5152e99702e22346bd2aaff06925715dcf1a00fc3",
-  "config_aarch64": "e5e5d73cb64189a18c4129d466545f81b8cb8fe8ceeacddeb39b6e73e1822fc1",
+  "config_aarch64": "48bcc7da9cb3da659da1d69ce9c07258553b2da398589464ceaf3e5893efdd7a",
```
Comment (christopherco) [inline SPECS/kernel/kernel.signatures.json:4]: Reminder to update the hash after finalizing the config changes

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/kernel/config_aarch64 [hunk 16]
```diff
@@ -30,8 +30,8 @@ CONFIG_SYSVIPC=y
 CONFIG_SYSVIPC_SYSCTL=y
 CONFIG_POSIX_MQUEUE=y
 CONFIG_POSIX_MQUEUE_SYSCTL=y
-# CONFIG_CROSS_MEMORY_ATTACH is not set
-# CONFIG_USELIB is not set
+CONFIG_CROSS_MEMORY_ATTACH=y
+CONFIG_USELIB=y
```
[the remark this answers is not in the document]
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated.

File: SPECS/kernel/config_aarch64 [hunk 17]
```diff
@@ -213,12 +224,12 @@ CONFIG_PERF_EVENTS=y
...
 CONFIG_VM_EVENT_COUNTERS=y
 CONFIG_SLUB_DEBUG=y
-# CONFIG_SLUB_MEMCG_SYSFS_ON is not set
+CONFIG_SLUB_MEMCG_SYSFS_ON=y
 # CONFIG_COMPAT_BRK is not set
 # CONFIG_SLAB is not set
 CONFIG_SLUB=y
 # CONFIG_SLOB is not set
-# CONFIG_SLAB_MERGE_DEFAULT is not set
+CONFIG_SLAB_MERGE_DEFAULT=y
```
[the remark this answers is not in the document]
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated.

File: SPECS/kernel/config_aarch64 (hunk 1 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated

File: SPECS/kernel/config_aarch64 (hunk 2 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated.

File: SPECS/kernel/config_aarch64 (hunk 3 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated

File: SPECS/kernel/config_aarch64 (hunk 4 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated

File: SPECS/kernel/config_aarch64 (hunk 13 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated.

File: SPECS/kernel/config_aarch64 (hunk 14 above)
Comment (chalamalasetty) [reply SPECS/kernel/config_aarch64]: updated.

File: SPECS/kernel/kernel.signatures.json (hunk 15 above)
Comment (chalamalasetty) [reply SPECS/kernel/kernel.signatures.json]: updated

File: SPECS/kernel/config_aarch64 (hunk 5 above)
Comment (christopherco) [reply SPECS/kernel/config_aarch64:163]: Discussed offline. May be required for boot. Will prune down in future PR.

File: SPECS/kernel/config_aarch64 (hunk 9 above)
Comment (christopherco) [reply SPECS/kernel/config_aarch64:600]: Discussed offline. May be required for boot. Will prune down in future PR.

File: SPECS/kernel/config_aarch64 (hunk 8 above)
Comment (christopherco) [reply SPECS/kernel/config_aarch64:638]: Discussed offline. May be required for boot. Will prune down in future PR.

File: SPECS/kernel/config_aarch64 (hunk 6 above)
Comment (christopherco) [reply SPECS/kernel/config_aarch64:991]: Discussed offline. May be required for boot. Will prune down in future PR.

**Review by christopherco — APPROVED** [review]

**Pull request merged**